index.js

```diff
@@ -130,18 +181,105 @@ function oponentMoveAnimation(el,lane){
           cloneEl.style.top = pos + "vh";
           cloneEl.style.left = lane;
           cloneEl.style.zIndex = Date.now() % 1000
-          if(pos >110){
+          var idOfCar = document.getElementById('player');
+          var carCurrentPostion = parseInt(idOfCar.style.left || 1);    //here the by default value of car_currentposition is changed to 1 insted of 0
+                                                                        //to handle the crash event in case user just starts the game and do not move the car
+
+          var hasCrachOccurred= crashOccurred(carCurrentPostion,lane,pos); //calls a method crashOccurred which check the events when crash can occur.
+            if(hasCrachOccurred)
+             {
+                cloneEl.parentElement.removeChild(cloneEl);
+                clearInterval(oponentMoveIntervalId);
+                oponentMoveIntervalId=null;
+                resetIntervals(); 
+             }
+             else if(flag)
+              { 
+                clearInterval(oponentMoveIntervalId);
+                oponentMoveIntervalId=null;
+                var reset=confirm("crash occurred");
+                if(reset)
+                {
+                  resetGame(cloneEl);
+                  return;
+                 //resetGame function is called that initialise the game;
+                }
+               }                  
+             else if(pos >110)
+             {
               cloneEl.parentElement.removeChild(cloneEl);
               clearInterval(oponentMoveIntervalId);
-          }
-      }, SPEED);
+             }
+      }, Speed);
+
+return;
+}
+
+function resetIntervals()
+{
+
+    clearInterval(speedIntervalId);
+    speedIntervalId=null;
+    clearInterval(generateCarId);  
+    generateCarId=null;
 }
 
+function resetGame(cloneEl)                        //this function is used to initialise the game
+{
+cloneEl.parentElement.removeChild(cloneEl);
+resetSpeed(50);
+flag=false;
+resetIntervals();
+}
+
+function crashOccurred(carCurrentPostion,lane,pos)
+{
+    //These if else statements are written to handle the crash event if the above
+    //contition satisfy crash will occur and all the interval will close
+    if(carCurrentPostion==posEleven && lane==middleLane && pos==posSeventySeven)           
+        {                                                                    
+         flag=true;
+          return true; 
+        }
+    else if(carCurrentPostion<=posSeven && lane==leftLane && (pos>=posSeventySeven && pos<=posNintyThree)) 
+        {       
+         flag=true; 
+         return true;
```

Comment:
why setting and using flag(global var) when this function can return required boolean.